api/src/test/java/org/apache/iceberg/util/TestCharSequenceMap.java

```diff
@@ -88,20 +92,38 @@ public void testClear() {
     assertThat(map).isEmpty();
   }
 
+  @Test
+  public void testKeySet() {
+    CharSequenceMap<String> map = CharSequenceMap.create();
+    map.put("key1", "value1");
+    map.put("key2", "value2");
```

Comment:
Shall we use a mix of different CharSequences as input? Like `StringBuilder` or `StringBuffer`?